### PR TITLE
[SYNTH-12985] Add properties to env variables

### PR DESCRIPTION
**Pull request by AntoineDona**
### What and why?

Many configuration properties were not configurable through environment variables:  `configPath`, `failOnCriticalErrors`, `failOnMissingTests`,`failOnTimeout`,`files`,`jUnitReport`,`latest`,`mobileApplicationId`,`publicIds`,`selectiveRerun`,`testSearchQuery`,`tunnel` and `versionName`. This PR adds support for configuring these properties through environment variables, all prefixed with `DATADOG_SYNTHETICS_`.

### How?

We fixed this problem by adding the the missing environment variables. In this PR, we also:

- Added `jUnitReport` property to the `Global Config`
- Added `testSearchQuery` to the `DEFAULT_COMMAND_CONFIG` as it was missing
- Created a utility function `toBoolean` to convert environment variable strings to boolean values, while handling edge cases (undefined values).

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

File: src/commands/synthetics/run-tests-command.ts [hunk 1]
```diff
@@ -195,10 +198,22 @@ export class RunTestsCommand extends Command {
   private async resolveConfig() {
     // Defaults < file < ENV < CLI
 
+    let overrideConfigPath = this.config.configPath
```
Comment (jakepruitt) [inline src/commands/synthetics/run-tests-command.ts:201]: I think this will never be defined, since the config file can't specify the path to... itself. I believe we want to start with the env config as the baseline, then put the CLI on top of it.

File: src/commands/synthetics/upload-application-command.ts [hunk 2]
```diff
@@ -106,7 +107,11 @@ export class UploadApplicationCommand extends Command {
       removeUndefinedValues({
         apiKey: process.env.DATADOG_API_KEY,
         appKey: process.env.DATADOG_APP_KEY,
+        configPath: process.env.DATADOG_SYNTHETICS_CONFIG_PATH,
```
Comment (jakepruitt) [inline src/commands/synthetics/upload-application-command.ts:110]: I believe you'll need to do a similar trick above -- we want to take the `process.env.DATADOG_SYNTHETICS_CONFIG_PATH` env var into consideration when calling `resolveConfigFromFile`.

File: src/commands/synthetics/run-tests-command.ts [hunk 3]
```diff
@@ -158,6 +154,13 @@ export class RunTestsCommand extends Command {
       return 1
     }
 
+    const reporters: Reporter[] = [new DefaultReporter(this)]
+    this.reporter = getReporter(reporters)
+
+    if (this.config.jUnitReport) {
+      reporters.push(new JUnitReporter(this))
```
Comment (jakepruitt) [inline src/commands/synthetics/run-tests-command.ts:161]: I think the `JUnitReporter` expects there to be a top-level `jUnitReport` available. We should change this behavior to rely on `this.config.jUnitReporter` instead

File: src/commands/synthetics/upload-application-command.ts (hunk 2 above)
Comment (AntoineDona) [reply src/commands/synthetics/upload-application-command.ts:110]: oh right, forgot about this one !

**Review by kayayarai — APPROVED** [review]
